[DOC-1411][2026.1.1] Updates to MCP server (#32720)
yugabyte/yugabyte-db: docs/content/stable/develop/AI/mcp-server.md

- Explore YugabyteDB data using natural language prompts
- Generate AI-powered visualizations from query results
- Enable LLMs to safely issue read-only SQL queries (and optionally guarded writes)
- Map authenticated users to database roles via OIDC identity mapping

> - Integrate instantly with tools like Claude Desktop, Cursor, and Windsurf

Reviewer: ```suggestion
- Integrate instantly with AI coding agents such as Claude Desktop, Cursor, and Windsurf
```

> Learn more about the YugabyteDB MCP Server:

Reviewer: Neither of these have examples for OIDC or guarded write integrations. @naveengupta-ybian

- [Build a Retail Agent with MCP Toolbox, YugabyteDB, and Google ADK](https://www.yugabyte.com/blog/build-a-retail-agent/)

> - [How to Integrate the YugabyteDB MCP Server with Visual Studio Code](https://www.yugabyte.com/blog/integrate-yugabytedb-mcp-server-with-vs-code/)

Reviewer: The VS Code blog has an outdated (and therefore erroneous) launch command. Need to update  src/server.py with uvx yugabytedb-mcp-server. @samiahmedsiddiqui @naveengupta-ybian

Author: Commented out old blogs

## Example: Claude Desktop

This tutorial walks you through using the YugabyteDB MCP Server to allow an AI application to access, query, analyze, and interpret data in your YugabyteDB database, using only natural language prompts.

The tutorial uses a YugabyteDB cluster running the [Northwind dataset](/stable/develop/sample-data/northwind/). You connect [Claude](https://claude.com/product/overview) to this database using MCP, and then explore it using natural language prompts.

### Prerequisites

- YugabyteDB {{<release "2025.2">}} or later
- Python 3.10+
- [uv](https://docs.astral.sh/uv/) (recommended) or pip / pipx
- [Claude Desktop](https://claude.ai/download)

### MCP tools

The server exposes three tools that MCP clients can call:

| Tool | Description |
| ---- | ----------- |
| `summarize_database` | Lists tables in a schema (default `public`) with columns and row counts. Read-only. |

> | `run_read_only_query` | Executes a SQL query inside `BEGIN READ ONLY` and returns rows as JSON. Read-only. |

Reviewer 2: Tested the example by following the instructions and checked that "run_read_only_query" is called in Claude Desktop.

    }
    ```

    **Via an installed script** (after `pipx install` or `uv tool install`):

    ```json
    {
        "mcpServers": {
            "yugabytedb-mcp": {
                "command": "yugabytedb-mcp",
                "env": {
                    "YUGABYTEDB_URL": "dbname=northwind host=localhost port=5433 user=yugabyte password=yugabyte"
                }
            }
        }
    }
    ```

    Replace the database user and password as appropriate.

    Configuration file locations:

    - macOS: `~/Library/Application Support/Claude/claude_desktop_config.json`
    - Windows: `%APPDATA%\Claude\claude_desktop_config.json`

1. Restart Claude to apply changes.

> The same configuration works with **Cursor** (Settings → MCP) and **Windsurf** (Settings → Cascade → MCP Servers).

Reviewer 2: ```suggestion
The same configuration works with **Cursor** (Settings > MCP) and **Windsurf** (Settings > Cascade > MCP Servers).
```

over the ~2-year period covered by the dataset.
```

## OIDC and per-user database roles

For shared or remote deployments (HTTP transport), you can enable OIDC (AWS Cognito or a generic OIDC provider) so each authenticated caller runs SQL under their own YugabyteDB role via `SET ROLE`.

> Identity mapping mirrors YSQL's native OIDC → role mapping (`ysql_ident_conf_csv` / `matching_claim_key`):

Reviewer 2: ```suggestion
Identity mapping mirrors YSQL's native OIDC role mapping (`ysql_ident_conf_csv` / `matching_claim_key`):
```

| Environment variable | Purpose |
| -------------------- | ------- |
| `YB_MCP_IDENTITY_CLAIM` | JWT claim used as the identity (default `email`). Supports dotted paths (for example, `realm_access.roles`) and Cognito-style keys (`cognito:groups`). |
| `YB_MCP_IDENTITY_TRANSFORM` | `none` or `strip_domain` when no map file is set. |

> | `YB_MCP_IDENTITY_MAP` | Path to a `pg_ident.conf`-style map file (the allowlist of claim value → DB role). |

Reviewer 2: ```suggestion
| `YB_MCP_IDENTITY_MAP` | Path to a `pg_ident.conf`-style map file (the allowlist of claim value maps to DB role). |
```

| `YB_MCP_IDENTITY_MAP_NAME` | Named map inside the file to apply (default `default`). |
| `YB_MCP_REQUIRE_ACCESS_TOKEN` | Cognito-only. When `true`, reject tokens with `token_use` other than `access`. Default `false` for backward compatibility. |

When `YB_MCP_IDENTITY_MAP` is set, unmapped claim values are rejected (`IdentityError`) — the map is the allowlist. JWT audience validation is always on for Cognito and OIDC.

> For full provider setup, map file format, worked examples (Cognito, Keycloak, Azure AD), and a migration checklist, see [OIDC.md](https://github.com/yugabyte/yugabytedb-mcp-server/blob/main/OIDC.md) in the MCP server repository.

Reviewer: Let's make sure we merge [PR10](https://github.com/yugabyte/yugabytedb-mcp-server/pull/10) before we merge this docs PR or this link will be dead @ddhodge

> For full provider setup, map file format, worked examples (Cognito, Keycloak, Azure AD), and a migration checklist, see [OIDC.md](https://github.com/yugabyte/yugabytedb-mcp-server/blob/main/OIDC.md) in the MCP server repository.

Reviewer 2: ```suggestion
For full provider setup, map file format, worked examples (Cognito, Keycloak, Azure AD), and a migration checklist, see [OIDC.md](https://github.com/yugabyte/yugabytedb-mcp-server/blob/main/OIDC.md) in the YugabyteDB MCP server repository.
```
